### ISSUE TITLE
Issue on page /tutorials/gs-ising.html

### ISSUE THREAD

**Issue by yiminghwang** [netket/netket#1998]
Hi there, 

I have a dummy question that I follow the tutorial to build a NN state for finding the ground energy of an Ising model where I follows the following codes attached in the tutorial:

class FFN(nnx.Module):

    def __init__(self, N: int, alpha: int = 1, *, rngs: nnx.Rngs):
        """
        Construct a Feed-Forward Neural Network with a single hidden layer.

        Args:
            N: The number of input nodes (number of spins in the chain).
            alpha: The density of the hidden layer. The hidden layer will have
                N*alpha nodes.
            rngs: The random number generator seed.
        """
        self.alpha = alpha

        # We define a linear (or dense) layer with `alpha` times the number of input nodes
        # as output nodes.
        # We must pass forward the rngs object to the dense layer.
        self.linear = nnx.Linear(in_features=N, out_features=alpha * N, rngs=rngs)

    def __call__(self, x: jax.Array):

        # we apply the linear layer to the input
        y = self.linear(x)

        # the non-linearity is a simple ReLu
        y = nnx.relu(y)

        # sum the output
        return jnp.sum(y, axis=-1)

if __name__ == "__main__":
    N = 20

    H, hi, eigvals = transverse_field_Ising(N)

    optimizer = nk.optimizer.Sgd(learning_rate=0.1)

    sampler = nk.sampler.MetropolisLocal(hi)

    model = FFN(N=N, alpha=1, rngs=nnx.Rngs(2))
    vstate = nk.vqs.MCState(sampler, model, n_samples=1008)

    # Notice the use, again of Stochastic Reconfiguration, which considerably improves the optimisation
    gs = nk.driver.VMC(
        H,
        optimizer,
        variational_state=vstate,
        preconditioner=nk.optimizer.SR(diag_shift=0.1),
    )

    n_iter = 300
    log = nk.logging.RuntimeLog()
    gs.run(n_iter=n_iter, out=log)
    ffn_energy = vstate.expect(H)
    error = abs((ffn_energy.mean - eigvals) / eigvals)
    print("Optimized energy and relative error: ", ffn_energy, error)

However, I got the following error about the initialization,

![Image](https://github.com/user-attachments/assets/ad2785c5-e795-42dc-8757-5177b6a712e9)

Is there some thing I missed?

P.S. my runtime environment: python=3.9.21 jax=0.4.30 jaxlib=0.4.30 flax=0.8.5 netket=3.13.0

**Comment by PhilipVinc** [comment]
The example you see is for the last version of netket, 3.16, because it uses `flax.nnx`.
You are  on 3.13.

That's because you are using python 3.9. Jax and NetKet have not been supporting this version for ~9 months now.
You should switch to a more recent version of python, or look at older versions of the tutorials.
But I strongly suggest to use a recent python (3.12 for example)

**Comment by yiminghwang** [comment]
Thanks !
